Environment_Engine/Query/Panels.cs

```diff
@@ -290,5 +290,121 @@ public static List<Panel> PanelsFromPhysical(this List<BH.oM.Physical.Elements.I
 
             return panels;
         }
+
+        [Description("Returns the floor panels of a space represented by Environment Panels")]
+        [Input("panelsAsSpace", "A collection of Environment Panels that represent a closed space")]
+        [Output("floorPanels", "BHoM Environment panel representing the floor of the space")]
+        public static List<Panel> FloorPanels(this List<Panel> panelsAsSpace)
+        {
+            //Find the panel(s) that are at the lowest point of the space...
+
+            double minZ = 1e10;
+            foreach (Panel panel in panelsAsSpace)
+            {
+                if (panel.MinimumLevel() == panel.MaximumLevel())
+                    minZ = Math.Min(minZ, panel.MinimumLevel());
+            }
+
+            List<Panel> floorPanels = panelsAsSpace.Where(x => x.MinimumLevel() == minZ && x.MaximumLevel() == minZ).ToList();
+
+            foreach (Panel panel in floorPanels)
+            {
+                if (panel.ConnectedSpaces.Where(x => x != "-1").ToList().Count == 1)
+                    panel.Type = PanelType.SlabOnGrade;
+                else if (panel.ConnectedSpaces.Where(x => x != "-1").ToList().Count == 2)
+                    panel.Type = PanelType.FloorInternal;
+
+                if (floorPanels.Count == 0)
```

Comment:
There is a failure in the logic here - this `if` statement will never execute because it is within the `for` loop. This `if` statement needs to be brought out of the `for` loop.

Reply:
done